bpf/sockops/bpf_redir.c

```diff
@@ -61,39 +61,10 @@ int bpf_redir_proxy(struct sk_msg_md *msg)
 	else
 		dstID = WORLD_ID;
 
-	info = lookup_ip4_remote_endpoint(key.sip4);
-	if (info != NULL && info->sec_label)
-		srcID = info->sec_label;
-	else
-		srcID = WORLD_ID;
-
 	verdict = policy_sk_egress(dstID, key.sip4, key.dport);
-	if (redirect_to_proxy(verdict)) {
-		struct proxy4_tbl_value value = {
-			.orig_daddr = key.dip4,
-			.orig_dport = key.dport,
-			.identity = srcID,
-		};
-		struct proxy4_tbl_key proxy_key = {
-			.saddr = key.sip4,
-			.sport = key.sport,
-			.dport = verdict,
-			.nexthdr = IPPROTO_TCP,
-		};
-		__be32 host_ip = IPV4_GATEWAY;
-
-		value.lifetime = bpf_ktime_get_sec() + PROXY_DEFAULT_LIFETIME;
-
-		key.dport = verdict;
-		key.dip4 = host_ip;
-
-		if (map_update_elem(&PROXY4_MAP, &proxy_key, &value, 0) < 0)
-			return SK_PASS;
-		err = msg_redirect_hash(msg, &SOCK_OPS_MAP, &key, flags);
-	} else if (!verdict) {
+	if (verdict != 0) {
```

Comment:
Comparing the previous version of this vs the current version of this, I'm confused.

Previously:

```
if (redirect_to_proxy(verdict)) {  // ie, if verdict is > 0
   ... update proxy4map
   ... msg_redirect_hash(...)
} else if (!verdict) { // ie, verdict is 0
   ... msg_redirect_hash(...)
}
```

Now:

```
if (verdict != 0) { // ie, < 0 or > 0
   ...
}
```

Reply:
Thanks for spotting this, changed this to:
```
	if (verdict >= 0) {
		msg_redirect_hash(msg, &SOCK_OPS_MAP, &key, flags);
	}
```

This should retain previous functionality.

Reply:
Is there some tests we can run against this? It should have shown the failure previously and show that it now works (unless there's something else hiding in the wings)